src/lib/transition_handler/validator.ml

```diff
@@ -33,16 +33,14 @@ module Make (Inputs : Inputs.With_unprocessed_transition_cache.S) :
       |> External_transition.Verified.protocol_state
     in
     let%bind () =
-      Result.ok_if_true
-        (Transition_frontier.find frontier hash |> Option.is_none)
-        ~error:`Duplicate
+      Option.fold (Transition_frontier.find frontier hash) ~init:Result.ok_unit
+        ~f:(fun _ _ -> Result.Error (`In_frontier hash) )
     in
     let%bind () =
-      Result.ok_if_true
-        (not
-           (Unprocessed_transition_cache.mem unprocessed_transition_cache
-              transition_with_hash))
-        ~error:`Duplicate
+      Option.fold
```

Comment:
I think there is a more canonical/readable alternative to using `Option.fold` like this (here and above):

```ocaml
Result.ok_if_true (Option.is_none your_option) ~error:your_error
```

For example, the above usage would become:

```ocaml
Result.ok_if_true (Option.is_none (Transition_frontier.find frontier hash)) ~error:(`In_frontier hash)
```